engine/src/main/scala/cromwell/jobstore/JobStoreWriterActor.scala

```diff
@@ -35,17 +37,25 @@ case class JobStoreWriterActor(jsd: JobStore,
       // Filter job completions that also have a corresponding workflow completion; these would just be
       // immediately deleted anyway.
       val jobCompletions = completions.toList collect { case j: JobCompletion if !completedWorkflowIds.contains(j.key.workflowId) => j }
-      val databaseAction = jsd.writeToDatabase(workflowCompletions, jobCompletions, batchSize)
+      val jobStoreAction: Future[Unit] = jsd.writeToDatabase(workflowCompletions, jobCompletions, batchSize)
+      val workflowStoreAction: Future[List[Int]] = Future.sequence {
+        completedWorkflowIds.map(workflowStoreAccess.deleteFromStore(_)).toList
+      }
+
+      val combinedAction: Future[Unit] = for {
+        _ <- jobStoreAction
+        _ <- workflowStoreAction
+      } yield ()
```

Comment:
To look into: what happens if we restart after all jobs completed but the workflow wasn't recorded? We should not lose all the jobs.